internal/app/thanos-notif/app.go

```diff
@@ -16,25 +21,192 @@ type Notifier interface {
+	ERC20WithdrawalFinalizedEventABI = "ERC20WithdrawalFinalized(address,address,address,address,uint256,bytes)"
+	DepositFinalizedEventABI         = "DepositFinalized(address,address,address,address,uint256,bytes)"
+	WithdrawalInitiatedEventABI      = "WithdrawalInitiated(address,address,address,address,uint256,bytes)"
+)
+
+const (
+	usdcSepolia = "0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238"
+	usdcMainnet = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"
+	tonSepolia  = "0xa30fe40285b8f5c0457dbc3b7c8a280373c40044"
+	tonMainnet  = "0x2be5e8c109e2197D077D13A82dAead6a9b3433C5"
 )
 
 type App struct {
 	cfg      *Config
 	notifier Notifier
 }
 
-func (app *App) ERC20TransferEvent(vLog *types.Log) {
-	log.GetLogger().Infow("Got ERC20TransferEvent", "event", vLog)
+func (app *App) ETHDepAndWithEvent(vLog *types.Log) {
+	log.GetLogger().Infow("Got ETH Deposit or Withdrawal Event", vLog)
+
+	txHash := vLog.TxHash
+	FromTo := common.HexToAddress(vLog.Topics[1].Hex())
+
+	// ETH deposit and withdrawal Amount
+	amountData := vLog.Data[:32]
+
+	decimals := 18
+	value := new(big.Int).SetBytes(amountData)
+	decimalFactor := new(big.Int).Exp(big.NewInt(10), big.NewInt(int64(decimals)), nil)
+	amountFloat := new(big.Float).SetInt(value)
+	amountFloat.Quo(amountFloat, new(big.Float).SetInt(decimalFactor))
+
+	Amount := strings.TrimRight(strings.TrimRight(amountFloat.Text('f', decimals+1), "0"), ".")
+
+	// Slack notify title and text
+	var title string
+	var text string
+
+	if common.HexToAddress(vLog.Topics[0].Hex()) == common.HexToAddress("0x35d79ab81f2b2017e19afb5c5571778877782d7a8786f5907f93b0f4702f4f23") {
+		title = fmt.Sprintf("[" + app.cfg.Network + "] [ETH Deposit Initialized]")
+		text = fmt.Sprintf("Tx: "+app.cfg.L1ExplorerUrl+"/tx/%s\nFrom: "+app.cfg.L1ExplorerUrl+"/address/%s\nTo: "+app.cfg.L2ExplorerUrl+"/address/%s\nAmount: %+v ETH", txHash, FromTo, FromTo, Amount)
+	} else if common.HexToAddress(vLog.Topics[0].Hex()) == common.HexToAddress("0x2ac69ee804d9a7a0984249f508dfab7cb2534b465b6ce1580f99a38ba9c5e631") {
+		title = fmt.Sprintf("[" + app.cfg.Network + "] [ETH Withdrawal Finalized]")
+		text = fmt.Sprintf("Tx: "+app.cfg.L1ExplorerUrl+"/tx/%s\nFrom: "+app.cfg.L2ExplorerUrl+"/address/%s\nTo: "+app.cfg.L1ExplorerUrl+"/address/%s\nAmount: %+v ETH", txHash, FromTo, FromTo, Amount)
+	} else {
+		title = "Unknown Event"
+	}
+
+	app.notifier.Notify(title, text)
+}
+
+func (app *App) ERC20DepAndWithEvent(vLog *types.Log) {
+	log.GetLogger().Infow("event", vLog)
+
+	var decimals int
+	var tokenSymbol string
+
+	tokenAddress := common.HexToAddress(vLog.Topics[1].Hex())
+
+	if tokenAddress == common.HexToAddress(usdcSepolia) || tokenAddress == common.HexToAddress(usdcMainnet) {
+		decimals = 6
+		tokenSymbol = "USDC"
+	} else if tokenAddress == common.HexToAddress(tonSepolia) || tokenAddress == common.HexToAddress(tonMainnet) {
+		decimals = 18
+		tokenSymbol = "TON"
+	} else {
+		decimals = 18
+		tokenSymbol = "ETH"
+	}
```

Comment:
Can you optimize it to a struct that likes

```go
type Token struct {
    Address common.Address
    Decimals uint64
    Symbol string
}
```

And create a map that stores the constant tokens

Reply:
@0x6e616d  Thank you for your feedback !!
I will edit it based on your advice. 🙇‍♀️